flutter/android/app/src/main/kotlin/com/carriez/flutter_hbb/FFIInitSingleton.kt

```diff
@@ -0,0 +1,17 @@
+package com.carriez.flutter_hbb
+
+import ffi.FFI
+
+import android.content.Context
+
+/**
+ * Static singleton class to initialize NDK contexts in FFI for hardware codecs
+ */
+
+enum class FFIInitSingleton {
+    INSTANCE;
+
+    fun init(ctx: Context) {
+        FFI.init(ctx)
+    }
+}
```

Comment:
I do not think this is correct, though I have never learned Kotlin.

Reply:
FFIInitSingleton.INSTANCE is singleton, it does mean init called once.

Reply:
Chatgpt can help you.
```
import android.content.Context
import java.util.concurrent.atomic.AtomicBoolean

/**
 * Static singleton class to initialize NDK contexts in FFI for hardware codecs
 */

enum class FFIInitSingleton {
    INSTANCE;

    private val initCalled = AtomicBoolean(false)

    fun init(ctx: Context) {
        if (initCalled.compareAndSet(false, true)) {
            FFI.init(ctx)
        } else {
            // Optionally, you can log or handle the case where init has already been called
            println("Init has already been called.")
        }
    }
}
```

Reply:
Yup I noticed :) Thanks